coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+The code coverage tool only collect code coverage for Go files
+
+
+#Design of Test Coverage Tool
+We pack the test coverage feature in a container, that is triggered to run by a CI/CD system such as prow, in response to Github events such as pulls and merges.The behavior varies in presubmit and postsubmit workflows, which is discussed in latter sections. 
+
+The tool takes input from three sources
+1. It runs test coverage profiling on target repository. 
+2. It reads previously stored post-submit code coverage profile from gcs bucket. The profile
+serves as a base of comparison for presubmit delta coverage.
+3. Variables passed through flags. Those variables include directory to run test coverage, file filters and threshold for desired coverage level.  
+
+Here is the step-by-step description of the pre-submit and post-submit workflows
+##Pre-submit workflow
+Runs code coverage tool to report coverage change in a new PR or updated PR
+1. Developer submit new commit to an open PR on github
+2. Matching pre-submit job is started 
+3. Generate coverage profile in artifacts directory
+4. Calculate coverage changes. Compare the coverage file generated in this cycle against the most
+ recent successful post-submit build. Coverage file for post-submit commits were generated in 
+ post-submit workflow and stored in gcs bucket
+5. Use PR data from github, git-attributes, as well as coverage change data calculated above, to 
+produce a list of files that we care about in the line-by-line coverage report. produce line by 
+line coverage html and add link to covbot report. Note that covbot is the robot github account 
+used to report code coverage change results. See Covbot section for more details.
+6. Let covbot post presubmit coverage on github, under that conversation of the PR. 
+
+##Post-submit workflow
+Produces & stores coverage profile for later presubmit jobs to compare against; 
+Produces periodical coverage result as input for TestGrid. Testgrid can use the data produced here to display coverage trend in a tabular or graphical way. 
+1. A PR is merged
+2. Post-submit job started
+3. Generate coverage profile. Completion marker generated upon successful run. Both stored
+ in artifacts directory.
+    - Completion marker is used by later pre-submit job when searching for a healthy and complete 
+    code coverage profile in the post-submit jobs
+    - Successfully generated coverage profile may be used as the basis of comparison for coverage change, 
+    as mentioned in pre-submit workflow
+4. Generate / store per-file coverage data
+    - Stores in the XML format, that is used by TestGrid, and dump it in artifacts directory
+
+##Locally running presubmit and post-submit workflows
+Both workflows may be triggered locally in command line, as long as all the required flags are 
+supplied correctly. In addition, the following env var needs to be set:
+- JOB_TYPE (one of 'presubmit', 'postsubmit', 'periodic', 'local-presubmit')
```

Comment:
`local-presubmit` is not a valid job type for Prow. Why does this tool consume this?

Reply:
This section talks about running the tool locally. Why does it need to be a valid job type for Prow?